src/d_iwad.c

```diff
@@ -345,11 +348,198 @@ static void CheckInstallRootPaths(void)
 
 // Check for Doom downloaded via Steam
 
+#define MAX_STEAM_LIBRARIES 16
+
+static char *steam_libraries[MAX_STEAM_LIBRARIES];
+static int num_steam_libraries = 0;
+
+static void AddSteamLibraryDir(char *dir)
+{
+    if (num_steam_libraries < MAX_STEAM_LIBRARIES)
+    {
+        steam_libraries[num_steam_libraries] = M_StringReplace(dir, "\\\\",
+                                                               "\\");
+        ++num_steam_libraries;
+    }
+}
+
+//Read whole of libraryfolders.vdf as string
+//silently fail is something goes wrong reading the file
+
+static char *GetSteamLibText(char *library_path)
+{
+    char *library_file_buffer;
+    size_t library_file_size;
+    size_t count;
+    char *probe;
+    FILE *library_file;
+
+    probe = M_FileCaseExists(library_path);
+
+    if (probe == NULL)
+    {
+        return NULL;
+    }
+
+    library_file = fopen(probe, "rb");
+
+    if (library_file == NULL)
+    {
+        return NULL;
+    }
+
+    library_file_size = M_FileLength(library_file);
+    library_file_buffer = malloc(library_file_size + 1);
+    count = fread(library_file_buffer, 1, library_file_size, library_file);
+    fclose(library_file);
+
+    if (library_file_size != count)
+    {
+        return NULL;
+    }
+
+    library_file_buffer[library_file_size] = '\0';
+    return library_file_buffer;
+}
+
+static char *GetNextSteamLibToken(char **lib_file_buffer)
+{
+    char *buffer;
+    char *token;
+    size_t i, buff_size;
+    boolean success;
+
+    token = NULL;               //a token is anything between "'s'
+    success = false;            //we want to avoid an incomplete token
+                                //so we use this instead of NULL check
+    buffer = *lib_file_buffer;
+    buff_size = strlen(buffer);
+
+    for (i=0; i<buff_size; ++i)
+    {
+        if (buffer[i] == '"')
+        {
+            //first " is the start of the token
+            if (token == NULL)
```

Comment:
It becomes simpler if you just do two loops: one to find the starting ", another to find the ending ".

Reply:
Thanks, this has helped.